FreeRTOS/Test/CMock/tasks/Makefile

```diff
@@ -0,0 +1,215 @@
+# Change according to what your  unit test directory is.
+# For example if testing queue.c your directory should be called queue
+# and the project name should be queue
+# if testing list.c your directory should be called list
+# and the project name should be list
+
+PROJECT          :=   tasks
+#
+# Test/CMock/tasks
+PROJECT_DIR      :=   $(abspath .)
+
+# List the dependency files you wish to mock
+MOCK_FILES_FP    :=   $(KERNEL_DIR)/include/timers.h
+MOCK_FILES_FP    +=   $(KERNEL_DIR)/include/list.h
+MOCK_FILES_FP    +=   $(KERNEL_DIR)/include/portable.h
+MOCK_FILES_FP    +=   $(PROJECT_DIR)/list_macros.h
+
+UNDEF_MOCKED_HEADER_MACROS := -UlistLIST_IS_EMPTY -UlistGET_OWNER_OF_HEAD_ENTRY \
```

Comment:
Is it really necessary to re-implement list.c?

Reply:
we are not, it is just that cmock can't detect macros
so i transformed these macro into function prototypes in order for cmock to detect and mock them

Reply:
My opinion is still that we should just include the actual list.c / list.h and not generate coverage for them.